### DISCUSSION TITLE
batch resize different dimension images to X

### DISCUSSION THREAD

**Discussion by Gunlaus** [ImageMagick/ImageMagick#2671] [Help]
Windows 10
ImageMagick 7
---
Greetings,

I have a folder with 100+ images with different dimensions and I want to batch resize all to 4x or 8x their size sampled to nearest neighbor to a specific output folder (4x H 4x W, for example). Is it possible? Would love some help with this, I'm a beginner with this app.

Thank you.

**Comment by snibgo** [comment]
First, find the command that does what you want to a single file. I suppose this would be:
```
magick in.png -sample 400% out.png
```
If you want 8x instead, then use 800%.

If you want both 4x and 8x, you can do it like this:

```
magick in.png ( +clone -sample 800% -write out1.png +delete ) -sample 400% out2.png
```
When you have a command that does what you want, either put it in a shell loop, or use "magick mogrify", eg:
```
magick mogrify -sample 400% *.png
```
If you use a BAT script, remember to double the percent signs: %%
